### PR TITLE
support client_info in Owin and add web app calls web API support

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Web;
+using Microsoft.Graph;
+
+namespace Microsoft.Identity.Web
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs:7]: We'll change this to extension methods on the Controller or ControllerBase.

File: src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    /// Extension methods to retrieve a Graph service or a token acquirer client from the HttpContext
+    /// </summary>
+    public static class HttpContextBaseExtensions
+    {
+        /// <summary>
+        /// Get the graph service client.
+        /// </summary>
+        /// <param name="httpContextBase"></param>
+        /// <returns></returns>
+        public static GraphServiceClient? GetGraphServiceClient(this HttpContextBase httpContextBase)
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs:19]: Same thing. Extension method on the controller.

File: src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            return AppBuilderExtension.ServiceProvider?.GetService(typeof(GraphServiceClient)) as GraphServiceClient;
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs:21]: We need to reconcile this with the ITokenAcquirerFactory

File: tests/aspnet-mvc/OwinWebApp/App_Start/Startup.Auth.cs [hunk 4]
```diff
@@ -16,6 +18,7 @@ public void ConfigureAuth(IAppBuilder app)
 
             app.AddMicrosoftIdentityWebApp(configureServices: services =>
             {
+                services.Configure<ConfidentialClientApplicationOptions>(options => { options.RedirectUri = "https://localhost:44386/"; });
```
Comment (jmprieur) [inline tests/aspnet-mvc/OwinWebApp/App_Start/Startup.Auth.cs:21]: We might want to improve this.

File: tests/aspnet-mvc/OwinWebApp/Web.config [hunk 5]
```diff
@@ -9,11 +9,6 @@
     <add key="webpages:Enabled" value="false"/>
     <add key="ClientValidationEnabled" value="true"/>
     <add key="UnobtrusiveJavaScriptEnabled" value="true"/>
-    <add key="ida:ClientId" value="4cb8c23b-0612-4b84-ae0f-03a0ff654e1f"/>
```
Comment (jmprieur) [inline tests/aspnet-mvc/OwinWebApp/Web.config:12]: This is also possible to keep these settings, but it's more complex to describe the certificates.
Can we comment these lines out (for the docs)

File: tests/aspnet-mvc/OwinWebApp/appsettings.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        "EnableCacheSynchronization": false,
+        "ClientCredentials": [
+            {
+                "SourceType": "KeyVault",
+                "KeyVaultUrl": "https://webappsapistests.vault.azure.net",
+                "KeyVaultCertificateName": "Self-Signed-5-5-22"
+            }
+        ]
+    },
+    "CalledApi": {
```
Comment (jmprieur) [inline tests/aspnet-mvc/OwinWebApp/appsettings.json:18]: Wasn't used

**Review by jmprieur — APPROVED** [review]

File: src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web.OWIN/HttpContextBaseExtensions.cs:7]: ok. will update.

File: tests/aspnet-mvc/OwinWebApp/App_Start/Startup.Auth.cs (hunk 4 above)
Comment (jennyf19) [reply tests/aspnet-mvc/OwinWebApp/App_Start/Startup.Auth.cs:21]: Yes. will fix.

File: tests/aspnet-mvc/OwinWebApp/Web.config (hunk 5 above)
Comment (jennyf19) [reply tests/aspnet-mvc/OwinWebApp/Web.config:12]: yes. will update.

File: tests/aspnet-mvc/OwinWebApp/appsettings.json (hunk 6 above)
Comment (jennyf19) [reply tests/aspnet-mvc/OwinWebApp/appsettings.json:18]: ack.

File: tests/aspnet-mvc/OwinWebApp/App_Start/Startup.Auth.cs (hunk 4 above)
Comment (jennyf19) [reply tests/aspnet-mvc/OwinWebApp/App_Start/Startup.Auth.cs:21]: let's discuss this one.

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks for addressing the feedback.

and for the user challenge!

**Pull request merged**